src/network/core/address.cpp

```diff
@@ -242,9 +242,16 @@ SOCKET NetworkAddress::Resolve(int family, int socktype, int flags, SocketList *
 		strecpy(this->hostname, fam == AF_INET ? "0.0.0.0" : "::", lastof(this->hostname));
 	}
 
+#ifdef __EMSCRIPTEN__
+	hints.ai_flags &= ~AI_ADDRCONFIG;
+#endif
+
 	int e = getaddrinfo(StrEmpty(this->hostname) ? nullptr : this->hostname, port_name, &hints, &ai);
 
-	if (reset_hostname) strecpy(this->hostname, "", lastof(this->hostname));
+#ifndef __EMSCRIPTEN__
+	if (reset_hostname)
+#endif
+		strecpy(this->hostname, "", lastof(this->hostname));
```

Comment:
This is not really following the coding style. Can't you force `reset_hostname` to `true` for Emscripten instead ?